sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/KalixRunner.scala

```diff
@@ -225,6 +223,16 @@ final class KalixRunner private[javasdk] (
 
     // start containers if application (only possible when running locally)
     dockerComposeUtils.foreach { dcu =>
+      // check out runtime container version before starting it
+      val outdatedVersions = dcu.collectOutdatedVersions(BuildInfo.runtimeVersion)
+      if (outdatedVersions.nonEmpty) {
+        log.warn(
+          "The docker-compose file '{}' contains outdated runtime image(s): ({}). It's recommended to update your image(s) to '{}'",
+          dcu.file,
+          outdatedVersions.mkString(","),
+          s"${BuildInfo.runtimeImage}:${BuildInfo.runtimeVersion}")
```

Comment:
Here we could check and recommend the version running in prod, but for the moment I'm limiting it to the version used to build the SDK.